e2e/tests/workspace/workspace-share-modal.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { LinkText, WorkspaceAccessLevel } from 'app/text-labels';
+import { findOrCreateWorkspace, findWorkspaceCard, signIn, signInWithAccessToken } from 'utils/test-utils';
+import WorkspaceAboutPage from 'app/page/workspace-about-page';
+import { makeWorkspaceName } from 'utils/str-utils';
+import { config } from 'resources/workbench-config';
+import WorkspacesPage from 'app/page/workspaces-page';
+import WorkspaceDataPage from 'app/page/workspace-data-page';
+
+describe('Workspace Share Modal', () => {
+  const assignAccess = [
+    { accessRole: WorkspaceAccessLevel.Writer, userEmail: config.writerUserName },
+    { accessRole: WorkspaceAccessLevel.Reader, userEmail: config.readerUserName }
+  ];
+
+  // Create new workspace with default CDR version
+  const workspace = makeWorkspaceName();
+
+  test.each(assignAccess)('Share %s', async (assign) => {
+    await signInWithAccessToken(page);
+    await findOrCreateWorkspace(page, { workspaceName: workspace });
+
+    const dataPage = new WorkspaceDataPage(page);
+    await dataPage.waitForLoad();
+
+    const shareWorkspaceModal = await dataPage.shareWorkspace();
```

Comment:
could you add a check before this to observe that the collaborator is not in the list yet?